### ISSUE TITLE
what are you finetuning?

### ISSUE THREAD

**Issue by franchesoni** [luca-medeiros/lightning-sam#3]
It isn't clear in the readme what's the finetuning you do. Is it SAM as an interactive image segmentation network what you're finetuning? This would mean you want good segmentations given some inputs. Or is it using the SAM encoder as backbone of a traditionally supervised segmentation network? You say you train only using bounding boxes, how do you generate the bounding boxes? I still don't quite get the idea, it'd be great if you could clarify. Thanks for the good work!

**Comment by luca-medeiros** [comment]
Hey @franchesoni, that's a very valid question! In general, the idea of fine-tuning is to leverage a pre-trained model and adapt it to a specific task or dataset. In the case of SAM, the network can be finetuned to generate better segmentations for a given task.

This repo is an experiment; I'm trying to see if fine-tuning SAM using bounding boxes as prompts would increase the IoU or improve the quality of the masks in general. The bounding boxes are not generated, one can use a COCO format dataset to fine-tune SAM for a specific task where SAM does not perform well (e.g., segmenting text on documents) and then use that model with interactive prompts just like SAM.

The main point is, if I fine-tune SAM on the COCO2017 dataset and compare the quality of the masks with the original model, and the fine-tuned model performs better, fine-tuning might help people trying to use SAM for specific tasks.

**Comment by luca-medeiros** [comment]
Guess I should update the readme with a clear explanation of what I'm trying to achieve and how I think this is relevant

**Comment by austinmw** [comment]
What about fine-tuning when you have segmentation ground truth?

**Comment by amteusch** [comment]
> What about fine-tuning when you have segmentation ground truth? It seems you do use/need segmentation GT masks and not only bounding boxes, or am I misunderstanding? https://github.com/luca-medeiros/lightning-sam/blob/main/src/train.py#L94
> 
> Also is this finetuning just the decoder, or the image and prompt encoders as well?

Looks like just the decoder, at least if you use the default config https://github.com/luca-medeiros/lightning-sam/blob/94fbe0d36f856ada8fcf42a7ca0c9a65a504ab8e/src/config.py#L20-L23

I am also interested to understand the answer to your other question about segmentation GT masks. My custom dataset has them and trains fine!

**Comment by luca-medeiros** [comment]
You are right! It uses the mask to fine tune the mask decoder, as per paper. When I mentioned bounding boxes, I mean as input prompt. The paper uses keypoints, bboxes, rough masks, text but this repo I'm only exploring the bbox prompt, for now.

**Comment by rbavery** [comment]
With the resulting fine tuned model, can you use it without input prompts to generate masks like the original model? Or can it only be used with interactive prompts? Thanks for releasing this!
